File: tower_of_hanoi.c
#include <stdio.h>

void towerOfHanoi(int n, char src, char helper, char dest) {
    if (n == 1) {
        printf("Move disk 1 from %c to %c\n", src, dest);
        return;
    }

    towerOfHanoi(n - 1, src, dest, helper);
    printf("Move disk %d from %c to %c\n", n, src, dest);
    towerOfHanoi(n - 1, helper, src, dest);
}

int main() {
    int numDisks = 7; // change this to the number of disks you want to solve for
    printf("Tower of Hanoi with %d disks:\n", numDisks);
    towerOfHanoi(numDisks, 'A', 'B', 'C');
    return 0;
}

File: TowerOfHanoi.js
function towerOfHanoi(n, source, auxiliary, destination) {
    if (n === 1) {
        console.log(`Move disk 1 from ${source} to ${destination}`);
        return;
    }

    towerOfHanoi(n - 1, source, destination, auxiliary);
    console.log(`Move disk ${n} from ${source} to ${destination}`);
    towerOfHanoi(n - 1, auxiliary, source, destination);
}

const numDisks = 3; //Change this to the number of disks you want to solve for
console.log(`Tower of Hanoi with ${numDisks} disks:`);
towerOfHanoi(numDisks, 'A', 'B', 'C');
